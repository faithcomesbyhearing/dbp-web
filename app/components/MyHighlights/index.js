/**
 *
 * MyHighlights
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
// import Link from 'next/link';
import SvgWrapper from '../SvgWrapper';
import LegacyLink from '../LegacyLink';

class MyHighlights extends React.PureComponent {
	constructor(props) {
		super(props);

		this.state = {
			selectedId: '',
			selectedColor: '',
		};
	}

	highlightIcon(color) {
		return (
			<span className={'color-icon'} style={{ backgroundColor: `${color}` }} />
		);
	}

	render() {
		const {
			highlights,
			getReference,
			deleteHighlights,
			toggleNotesModal,
			startUpdateProcess,
		} = this.props;

		const handleClick = (newUrl) =>
			(e) => {
				if (Router.asPath === newUrl) {
					e.preventDefault(); // Stop navigation if it's the same URL
					toggleNotesModal(); // Only toggle modal
				}
			};

		return highlights.map((highlight) => (
			<div
				key={`${highlight.id}_${highlight.highlighted_color}`}
				id={`${highlight.id}_${highlight.highlighted_color}`}
				className={'highlight-item'}
			>
				<LegacyLink
					as={`/bible/${highlight.bible_id}/${highlight.book_id}/${
						highlight.chapter
					}`}
					href={`/bible/${highlight.bible_id}/${highlight.book_id}/${
						highlight.chapter
					}`}
				>
					<button
						onClick={toggleNotesModal}
						className="list-item"
						type="button"
					>
						<div className="title-text">
							<h4 className="title">{getReference(highlight)}</h4>
							<h4 className={'text'}>{highlight.bible_id}</h4>
						</div>
					</button>
				</LegacyLink>
				<button
					id={`${highlight.id}_${highlight.highlighted_color}-edit`}
					className={'edit-color'}
					tabIndex={0}
					type={'button'}
					onClick={() =>
						startUpdateProcess({
							id: highlight.id,
							color: highlight.highlighted_color,
						})
					}
				>
					{this.highlightIcon(highlight.highlighted_color)}
					<span>Edit</span>
				</button>
				<div className={'delete-highlight'}>
					<SvgWrapper
						className={'icon'}
						svgid={'delete'}
						id={`${highlight.id}_${highlight.highlighted_color}-delete`}
						onClick={() => deleteHighlights({ ids: [highlight.id] })}
					/>
					<span>Delete</span>
				</div>
			</div>
		));
	}
}

MyHighlights.propTypes = {
	highlights: PropTypes.array,
	getReference: PropTypes.func,
	deleteHighlights: PropTypes.func,
	toggleNotesModal: PropTypes.func,
	startUpdateProcess: PropTypes.func,
};

export default MyHighlights;
