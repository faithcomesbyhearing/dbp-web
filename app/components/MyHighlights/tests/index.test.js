import React from 'react';
import { render } from '@testing-library/react';
import MyHighlights from '..';
import {
	getHighlightReference,
	highlights,
} from '../../../utils/testUtils/notebookFunctions';

const props = {
	highlights,
	getReference: jest.fn((params) => getHighlightReference(params)),
	toggleNotesModal: jest.fn(),
	startUpdateProcess: jest.fn(),
	deleteHighlights: jest.fn(),
};

describe('<MyHighlights /> component', () => {
	it('should match snapshot with expected props', () => {
		const { asFragment } = render(<MyHighlights {...props} />);
		expect(asFragment()).toMatchSnapshot();
	});
});
