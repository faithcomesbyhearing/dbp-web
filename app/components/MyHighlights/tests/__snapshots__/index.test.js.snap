// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MyHighlights /> component should match snapshot with expected props 1`] = `
[
  <div
    className="highlight-item"
    id="24440733_rgba(173,221,121,0.7)"
  >
    <button
      className="list-item"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      <div
        className="title-text"
      >
        <h4
          className="title"
        >
          EPH - 4:14 - (ENGNAB)
        </h4>
        <h4
          className="text"
        >
          ENGNAB
        </h4>
      </div>
    </button>
    <button
      className="edit-color"
      id="24440733_rgba(173,221,121,0.7)-edit"
      onClick={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="color-icon"
        style={
          {
            "backgroundColor": "rgba(173,221,121,0.7)",
          }
        }
      />
      <span>
        Edit
      </span>
    </button>
    <div
      className="delete-highlight"
    >
      <svg
        className="icon"
        id="24440733_rgba(173,221,121,0.7)-delete"
        onClick={[Function]}
      >
        <use
          xlinkHref="/svglist.svg#delete"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
      <span>
        Delete
      </span>
    </div>
  </div>,
  <div
    className="highlight-item"
    id="25827775_rgba(252,230,0,0.5)"
  >
    <button
      className="list-item"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      <div
        className="title-text"
      >
        <h4
          className="title"
        >
          MAT - 3:3 - (ENGESV)
        </h4>
        <h4
          className="text"
        >
          ENGESV
        </h4>
      </div>
    </button>
    <button
      className="edit-color"
      id="25827775_rgba(252,230,0,0.5)-edit"
      onClick={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="color-icon"
        style={
          {
            "backgroundColor": "rgba(252,230,0,0.5)",
          }
        }
      />
      <span>
        Edit
      </span>
    </button>
    <div
      className="delete-highlight"
    >
      <svg
        className="icon"
        id="25827775_rgba(252,230,0,0.5)-delete"
        onClick={[Function]}
      >
        <use
          xlinkHref="/svglist.svg#delete"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
      <span>
        Delete
      </span>
    </div>
  </div>,
  <div
    className="highlight-item"
    id="25827776_rgba(84,185,72,0.5)"
  >
    <button
      className="list-item"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      <div
        className="title-text"
      >
        <h4
          className="title"
        >
          MRK - 16:5 - (ENGESV)
        </h4>
        <h4
          className="text"
        >
          ENGESV
        </h4>
      </div>
    </button>
    <button
      className="edit-color"
      id="25827776_rgba(84,185,72,0.5)-edit"
      onClick={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="color-icon"
        style={
          {
            "backgroundColor": "rgba(84,185,72,0.5)",
          }
        }
      />
      <span>
        Edit
      </span>
    </button>
    <div
      className="delete-highlight"
    >
      <svg
        className="icon"
        id="25827776_rgba(84,185,72,0.5)-delete"
        onClick={[Function]}
      >
        <use
          xlinkHref="/svglist.svg#delete"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
      <span>
        Delete
      </span>
    </div>
  </div>,
  <div
    className="highlight-item"
    id="25827777_rgba(84,185,72,0.5)"
  >
    <button
      className="list-item"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      <div
        className="title-text"
      >
        <h4
          className="title"
        >
          MAT - 1:4 - (ENGESV)
        </h4>
        <h4
          className="text"
        >
          ENGESV
        </h4>
      </div>
    </button>
    <button
      className="edit-color"
      id="25827777_rgba(84,185,72,0.5)-edit"
      onClick={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="color-icon"
        style={
          {
            "backgroundColor": "rgba(84,185,72,0.5)",
          }
        }
      />
      <span>
        Edit
      </span>
    </button>
    <div
      className="delete-highlight"
    >
      <svg
        className="icon"
        id="25827777_rgba(84,185,72,0.5)-delete"
        onClick={[Function]}
      >
        <use
          xlinkHref="/svglist.svg#delete"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
      <span>
        Delete
      </span>
    </div>
  </div>,
  <div
    className="highlight-item"
    id="25827778_rgba(80,165,220,0.5)"
  >
    <button
      className="list-item"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      <div
        className="title-text"
      >
        <h4
          className="title"
        >
          MAT - 1:17 - (ENGESV)
        </h4>
        <h4
          className="text"
        >
          ENGESV
        </h4>
      </div>
    </button>
    <button
      className="edit-color"
      id="25827778_rgba(80,165,220,0.5)-edit"
      onClick={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="color-icon"
        style={
          {
            "backgroundColor": "rgba(80,165,220,0.5)",
          }
        }
      />
      <span>
        Edit
      </span>
    </button>
    <div
      className="delete-highlight"
    >
      <svg
        className="icon"
        id="25827778_rgba(80,165,220,0.5)-delete"
        onClick={[Function]}
      >
        <use
          xlinkHref="/svglist.svg#delete"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
      <span>
        Delete
      </span>
    </div>
  </div>,
  <div
    className="highlight-item"
    id="25827779_rgba(252,230,0,0.5)"
  >
    <button
      className="list-item"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
      type="button"
    >
      <div
        className="title-text"
      >
        <h4
          className="title"
        >
          MAT - 1:6 - (ENGESV)
        </h4>
        <h4
          className="text"
        >
          ENGESV
        </h4>
      </div>
    </button>
    <button
      className="edit-color"
      id="25827779_rgba(252,230,0,0.5)-edit"
      onClick={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="color-icon"
        style={
          {
            "backgroundColor": "rgba(252,230,0,0.5)",
          }
        }
      />
      <span>
        Edit
      </span>
    </button>
    <div
      className="delete-highlight"
    >
      <svg
        className="icon"
        id="25827779_rgba(252,230,0,0.5)-delete"
        onClick={[Function]}
      >
        <use
          xlinkHref="/svglist.svg#delete"
          xmlnsXlink="http://www.w3.org/1999/xlink"
        />
      </svg>
      <span>
        Delete
      </span>
    </div>
  </div>,
]
`;
